test/FunctionalTests/Microsoft.OData.Edm.Tests/Csdl/CsdlWriterTests.cs

```diff
@@ -961,6 +961,41 @@ public void ShouldWriteAnnotationForEnumMember()
             Assert.Equal(expected, csdlStr);
         }
 
+        [Fact]
+        public void CanWritePropertyWithCoreTypeDefinitionButValidationFailed()
+        {
+            string expected =
+            "<?xml version=\"1.0\" encoding=\"utf-16\"?>" +
+            "<edmx:Edmx Version=\"4.0\" xmlns:edmx=\"http://docs.oasis-open.org/odata/ns/edmx\">" +
+              "<edmx:DataServices>" +
+                "<Schema Namespace=\"NS\" xmlns=\"http://docs.oasis-open.org/odata/ns/edm\">" +
+                  "<ComplexType Name=\"Complex\">" +
+                    "<Property Name=\"ModifiedDate\" Type=\"Org.OData.Core.V1.LocalDateTime\" />" +
+                    "<Property Name=\"QualifiedName\" Type=\"Org.OData.Core.V1.QualifiedTypeName\" />" +
+                  "</ComplexType>" +
+                "</Schema>" +
+              "</edmx:DataServices>" +
+            "</edmx:Edmx>";
+
+            EdmModel model = new EdmModel();
+            var localDateTime = model.FindType("Org.OData.Core.V1.LocalDateTime") as IEdmTypeDefinition;
+            Assert.NotNull(localDateTime);
+
+            var qualifiedTypeName = model.FindType("Org.OData.Core.V1.QualifiedTypeName") as IEdmTypeDefinition;
+            Assert.NotNull(qualifiedTypeName);
+
+            EdmComplexType type = new EdmComplexType("NS", "Complex");
+            type.AddStructuralProperty("ModifiedDate", new EdmTypeDefinitionReference(localDateTime, true));
+            type.AddStructuralProperty("QualifiedName", new EdmTypeDefinitionReference(qualifiedTypeName, true));
+
+            model.AddElement(type);
+            IEnumerable<EdmError> errors;
+            Assert.False(model.Validate(out errors));
+            Assert.Equal(2, errors.Count());

```

Comment:
>2 [](start = 25, length = 1)

Why does validation fail for LocalDateTime, if it's added to the Core vocabulary?

Reply:
Failed for the Edm validation rule, the error message is:
`
{InvalidPropertyType : A property cannot be of type 'TypeDefinition'. The property type must be a complex, a primitive, an enum or an untyped type, or a collection of complex, primitive, or enum types.}
`
I thought that's a wrong validatation rule. I also sent it through Email to you. Please take a look.